lib/search.rb

```diff
@@ -1106,6 +1106,7 @@ def self.to_tsquery(ts_config: nil, term:, joiner: nil)
   end
 
   def self.set_tsquery_weight_filter(term, weight_filter)
+    term = term.gsub(/\\+$/, '') # Remove any trailing '\' to avoid quote escaping issues
     term = term.gsub("'", "''")
     "''#{PG::Connection.escape_string(term)}'':*#{weight_filter}"
```

Comment:
the amount of edge cases on top of simply using `escape_string` are a bit worrying here. Why is this only an issue for trailing slash? 

Does quote_literal solve this as well more cleanly? 

https://dba.stackexchange.com/questions/135030/filtering-special-characters-in-to-tsquery

Reply:
Oh possibly! I'll play around with it and see what I can figure out.

Reply:
@tgxworld could you help out here actually? 

I took a pass at adding quote_literal, but I'm not able to get it to work and keep running into pg syntax errors.

```
REPLACE(TO_TSQUERY('english', QUOTE_LITERAL(QUOTE_LITERAL('''Title with trailing slashes\\'':*A & ''no body'':*B')))::text, '&', '|')::tsquery
```

```
PG::SyntaxError: ERROR:  syntax error in tsquery: "E'E''''''Title with trailing slashes\\\\\\\\'''':*A & ''''no body'''':*B'''"
```


Another option from https://www.postgresql.org/docs/current/textsearch-controls.html#TEXTSEARCH-PARSING-QUERIES is to use plainto_tsquery but the docs say

> Note that plainto_tsquery will not recognize tsquery operators, weight labels, or prefix-match labels in its input:

And we are using some of those, so I'm not exactly sure how to work around them to get plainto_tsquery to work.

Reply:
This is also an issue for trailing question marks.